openssl: 1.1.0 -> 1.1.1

alyssais opened the pull request:
###### Motivation for this change

From https://www.openssl.org/blog/blog/2018/09/11/release111/:

> OpenSSL 1.1.1 is API and ABI compliant with OpenSSL 1.1.0 so most applications that work with 1.1.0 can gain many of the benefits of TLSv1.3 simply by dropping in the new OpenSSL version.

Added `openssl_1_0` and `openssl_1_1` aliases because that's the level of granularity packages almost certainly want, and will make future upgrades like this much easier.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

alyssais committed:
openssl: 1.1.0 -> 1.1.1

timokau commented:
If it is "API and ABI compliant with 1.1.0", why do we need the aliases? Are there actual cases where 1.1.1 causes a failure?

alyssais commented:
Not that I know of, but didn't want to break a convention by removing the `1_0_2` alias.

timokau commented:
Oh I just  noticed that you actually did remove the `1_1_0` version, I missed that at first.

The commit message is a bit confusing though. You don't actually update openssl,  you just update the dependency for a few versions right? Or am I misreading that?

alyssais commented:
I updated OpenSSL from 1.1.0 to 1.1.1. However, `nixpkgs.openssl` is the 1.0 channel. Both channels are simultaneously packaged by nixpkgs. So, I updated the 1.1 channel from 1.1.0 to 1.1.1.

alyssais commented on pkgs/development/libraries/openssl/default.nix:
```diff
@@ -125,9 +125,9 @@ in {
-    sha256 = "16fgaf113p6s5ixw227sycvihh3zx6f6rf0hvjjhxk68m12cigzb";
+  openssl_1_1_1 = common {
+    version = "1.1.1";
+    sha256 = "0gbab2fjgms1kx5xjvqx8bxhr98k4r8l2fa8vw7kvh491xd8fdi8";
```
Here is the actual upgrade.

timokau replied:
Right, thanks.

timokau commented:
LGTM assuming there are no build failures

@GrahamcOfBorg build ldns nodejs-10_x tor

GrahamcOfBorg commented:
<!--REQUEST_ID=eb0212a7-0ab4-4a91-bac8-ae7cbf825160-->
Failure on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.46524&attempt_id=3ceed613-ecba-440e-93a4-9f682e2f9fec)

Attempted: ldns, nodejs-10_x, tor

<details><summary>Partial log (click to expand)</summary><p>

```
Operating system: x86_64-whatever-linux2
./config: eval: line 1: /usr/bin/env: not found
building
build flags: -j2 -l2 SHELL=/nix/store/czx8vkrb9jdgjyz8qfksh10vrnqa723l-bash-4.4-p23/bin/bash MANDIR=\$\(man\)/share/man
make: *** No targets specified and no makefile found.  Stop.
builder for '/nix/store/r4da46wj7zk26vwwwfnc4gvp0n9axbx3-openssl-1.1.1.drv' failed with exit code 2
cannot build derivation '/nix/store/pqx6x1asb5qp6if2nipny8nmd0jlxdid-ldns-1.7.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/3h5q2czc35bzyvnri5m9zybcq5j4kdpx-nodejs-10.9.0.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/gzr81b0xzrvmmsdq8lsgvdv7zjq39ibk-tor-0.3.4.8.drv': 1 dependencies couldn't be built
error: build of '/nix/store/3h5q2czc35bzyvnri5m9zybcq5j4kdpx-nodejs-10.9.0.drv', '/nix/store/gzr81b0xzrvmmsdq8lsgvdv7zjq39ibk-tor-0.3.4.8.drv', '/nix/store/pqx6x1asb5qp6if2nipny8nmd0jlxdid-ldns-1.7.0.drv' failed
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=eb0212a7-0ab4-4a91-bac8-ae7cbf825160-->
Timed out, unknown build status on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.46524&attempt_id=b9fca3b7-d950-4d58-a49b-c15d956653ca)

Attempted: nodejs-10_x

The following builds were skipped because they don't evaluate on x86_64-darwin: ldns, tor

<details><summary>Partial log (click to expand)</summary><p>

```
  clang++ '-DV8_GYP_BUILD' '-DV8_TYPED_ARRAY_MAX_SIZE_IN_HEAP=0' '-D_DARWIN_USE_64_BIT_INODE=1' '-DV8_TARGET_ARCH_X64' '-DV8_EMBEDDER_STRING="-node.14"' '-DENABLE_DISASSEMBLER' '-DV8_PROMISE_INTERNAL_FIELD_COUNT' '-Dv8_promise_internal_field_count' '-DV8_INTL_SUPPORT' '-DV8_CONCURRENT_MARKING' '-DDISABLE_UNTRUSTED_CODE_MITIGATIONS' '-DICU_UTIL_DATA_IMPL=ICU_UTIL_DATA_STATIC' '-DUCONFIG_NO_SERVICE=1' '-DUCONFIG_NO_REGULAR_EXPRESSIONS=1' '-DU_ENABLE_DYLOAD=0' '-DU_STATIC_IMPLEMENTATION=1' '-DU_HAVE_STD_STRING=1' '-DUCONFIG_NO_BREAK_ITERATION=0' -I../deps/v8 -I../. -I/private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj/gen -I/nix/store/j54sy9zx13nmkcbic1g1dyhha4h1nnwl-zlib-1.2.11-dev/include -I/nix/store/1gadgxvbf1xdiffs9p9fz8xvb8hwx7ia-libuv-1.21.0/include -I/nix/store/b069mlwx07cm18p53y69p4asr2i81dym-openssl-1.1.1-dev/include -I../deps/v8/include -I../deps/icu-small/source/i18n -I../deps/icu-small/source/common  -O3 -gdwarf-2 -fstrict-aliasing -mmacosx-version-min=10.7 -arch x86_64 -Wall -Wendif-labels -W -Wno-unused-parameter -std=gnu++1y -stdlib=libc++ -fno-rtti -fno-exceptions -fno-threadsafe-statics -fno-strict-aliasing -MMD -MF /private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/.deps//private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj.target/v8_base/deps/v8/src/snapshot/serializer-common.o.d.raw   -c -o /private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj.target/v8_base/deps/v8/src/snapshot/serializer-common.o ../deps/v8/src/snapshot/serializer-common.cc
  clang++ '-DV8_GYP_BUILD' '-DV8_TYPED_ARRAY_MAX_SIZE_IN_HEAP=0' '-D_DARWIN_USE_64_BIT_INODE=1' '-DV8_TARGET_ARCH_X64' '-DV8_EMBEDDER_STRING="-node.14"' '-DENABLE_DISASSEMBLER' '-DV8_PROMISE_INTERNAL_FIELD_COUNT' '-Dv8_promise_internal_field_count' '-DV8_INTL_SUPPORT' '-DV8_CONCURRENT_MARKING' '-DDISABLE_UNTRUSTED_CODE_MITIGATIONS' '-DICU_UTIL_DATA_IMPL=ICU_UTIL_DATA_STATIC' '-DUCONFIG_NO_SERVICE=1' '-DUCONFIG_NO_REGULAR_EXPRESSIONS=1' '-DU_ENABLE_DYLOAD=0' '-DU_STATIC_IMPLEMENTATION=1' '-DU_HAVE_STD_STRING=1' '-DUCONFIG_NO_BREAK_ITERATION=0' -I../deps/v8 -I../. -I/private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj/gen -I/nix/store/j54sy9zx13nmkcbic1g1dyhha4h1nnwl-zlib-1.2.11-dev/include -I/nix/store/1gadgxvbf1xdiffs9p9fz8xvb8hwx7ia-libuv-1.21.0/include -I/nix/store/b069mlwx07cm18p53y69p4asr2i81dym-openssl-1.1.1-dev/include -I../deps/v8/include -I../deps/icu-small/source/i18n -I../deps/icu-small/source/common  -O3 -gdwarf-2 -fstrict-aliasing -mmacosx-version-min=10.7 -arch x86_64 -Wall -Wendif-labels -W -Wno-unused-parameter -std=gnu++1y -stdlib=libc++ -fno-rtti -fno-exceptions -fno-threadsafe-statics -fno-strict-aliasing -MMD -MF /private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/.deps//private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj.target/v8_base/deps/v8/src/snapshot/serializer.o.d.raw   -c -o /private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj.target/v8_base/deps/v8/src/snapshot/serializer.o ../deps/v8/src/snapshot/serializer.cc
  clang++ '-DV8_GYP_BUILD' '-DV8_TYPED_ARRAY_MAX_SIZE_IN_HEAP=0' '-D_DARWIN_USE_64_BIT_INODE=1' '-DV8_TARGET_ARCH_X64' '-DV8_EMBEDDER_STRING="-node.14"' '-DENABLE_DISASSEMBLER' '-DV8_PROMISE_INTERNAL_FIELD_COUNT' '-Dv8_promise_internal_field_count' '-DV8_INTL_SUPPORT' '-DV8_CONCURRENT_MARKING' '-DDISABLE_UNTRUSTED_CODE_MITIGATIONS' '-DICU_UTIL_DATA_IMPL=ICU_UTIL_DATA_STATIC' '-DUCONFIG_NO_SERVICE=1' '-DUCONFIG_NO_REGULAR_EXPRESSIONS=1' '-DU_ENABLE_DYLOAD=0' '-DU_STATIC_IMPLEMENTATION=1' '-DU_HAVE_STD_STRING=1' '-DUCONFIG_NO_BREAK_ITERATION=0' -I../deps/v8 -I../. -I/private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj/gen -I/nix/store/j54sy9zx13nmkcbic1g1dyhha4h1nnwl-zlib-1.2.11-dev/include -I/nix/store/1gadgxvbf1xdiffs9p9fz8xvb8hwx7ia-libuv-1.21.0/include -I/nix/store/b069mlwx07cm18p53y69p4asr2i81dym-openssl-1.1.1-dev/include -I../deps/v8/include -I../deps/icu-small/source/i18n -I../deps/icu-small/source/common  -O3 -gdwarf-2 -fstrict-aliasing -mmacosx-version-min=10.7 -arch x86_64 -Wall -Wendif-labels -W -Wno-unused-parameter -std=gnu++1y -stdlib=libc++ -fno-rtti -fno-exceptions -fno-threadsafe-statics -fno-strict-aliasing -MMD -MF /private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/.deps//private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj.target/v8_base/deps/v8/src/snapshot/snapshot-common.o.d.raw   -c -o /private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj.target/v8_base/deps/v8/src/snapshot/snapshot-common.o ../deps/v8/src/snapshot/snapshot-common.cc
  clang++ '-DV8_GYP_BUILD' '-DV8_TYPED_ARRAY_MAX_SIZE_IN_HEAP=0' '-D_DARWIN_USE_64_BIT_INODE=1' '-DV8_TARGET_ARCH_X64' '-DV8_EMBEDDER_STRING="-node.14"' '-DENABLE_DISASSEMBLER' '-DV8_PROMISE_INTERNAL_FIELD_COUNT' '-Dv8_promise_internal_field_count' '-DV8_INTL_SUPPORT' '-DV8_CONCURRENT_MARKING' '-DDISABLE_UNTRUSTED_CODE_MITIGATIONS' '-DICU_UTIL_DATA_IMPL=ICU_UTIL_DATA_STATIC' '-DUCONFIG_NO_SERVICE=1' '-DUCONFIG_NO_REGULAR_EXPRESSIONS=1' '-DU_ENABLE_DYLOAD=0' '-DU_STATIC_IMPLEMENTATION=1' '-DU_HAVE_STD_STRING=1' '-DUCONFIG_NO_BREAK_ITERATION=0' -I../deps/v8 -I../. -I/private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj/gen -I/nix/store/j54sy9zx13nmkcbic1g1dyhha4h1nnwl-zlib-1.2.11-dev/include -I/nix/store/1gadgxvbf1xdiffs9p9fz8xvb8hwx7ia-libuv-1.21.0/include -I/nix/store/b069mlwx07cm18p53y69p4asr2i81dym-openssl-1.1.1-dev/include -I../deps/v8/include -I../deps/icu-small/source/i18n -I../deps/icu-small/source/common  -O3 -gdwarf-2 -fstrict-aliasing -mmacosx-version-min=10.7 -arch x86_64 -Wall -Wendif-labels -W -Wno-unused-parameter -std=gnu++1y -stdlib=libc++ -fno-rtti -fno-exceptions -fno-threadsafe-statics -fno-strict-aliasing -MMD -MF /private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/.deps//private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj.target/v8_base/deps/v8/src/snapshot/snapshot-source-sink.o.d.raw   -c -o /private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj.target/v8_base/deps/v8/src/snapshot/snapshot-source-sink.o ../deps/v8/src/snapshot/snapshot-source-sink.cc
  clang++ '-DV8_GYP_BUILD' '-DV8_TYPED_ARRAY_MAX_SIZE_IN_HEAP=0' '-D_DARWIN_USE_64_BIT_INODE=1' '-DV8_TARGET_ARCH_X64' '-DV8_EMBEDDER_STRING="-node.14"' '-DENABLE_DISASSEMBLER' '-DV8_PROMISE_INTERNAL_FIELD_COUNT' '-Dv8_promise_internal_field_count' '-DV8_INTL_SUPPORT' '-DV8_CONCURRENT_MARKING' '-DDISABLE_UNTRUSTED_CODE_MITIGATIONS' '-DICU_UTIL_DATA_IMPL=ICU_UTIL_DATA_STATIC' '-DUCONFIG_NO_SERVICE=1' '-DUCONFIG_NO_REGULAR_EXPRESSIONS=1' '-DU_ENABLE_DYLOAD=0' '-DU_STATIC_IMPLEMENTATION=1' '-DU_HAVE_STD_STRING=1' '-DUCONFIG_NO_BREAK_ITERATION=0' -I../deps/v8 -I../. -I/private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj/gen -I/nix/store/j54sy9zx13nmkcbic1g1dyhha4h1nnwl-zlib-1.2.11-dev/include -I/nix/store/1gadgxvbf1xdiffs9p9fz8xvb8hwx7ia-libuv-1.21.0/include -I/nix/store/b069mlwx07cm18p53y69p4asr2i81dym-openssl-1.1.1-dev/include -I../deps/v8/include -I../deps/icu-small/source/i18n -I../deps/icu-small/source/common  -O3 -gdwarf-2 -fstrict-aliasing -mmacosx-version-min=10.7 -arch x86_64 -Wall -Wendif-labels -W -Wno-unused-parameter -std=gnu++1y -stdlib=libc++ -fno-rtti -fno-exceptions -fno-threadsafe-statics -fno-strict-aliasing -MMD -MF /private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/.deps//private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj.target/v8_base/deps/v8/src/snapshot/startup-deserializer.o.d.raw   -c -o /private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj.target/v8_base/deps/v8/src/snapshot/startup-deserializer.o ../deps/v8/src/snapshot/startup-deserializer.cc
  clang++ '-DV8_GYP_BUILD' '-DV8_TYPED_ARRAY_MAX_SIZE_IN_HEAP=0' '-D_DARWIN_USE_64_BIT_INODE=1' '-DV8_TARGET_ARCH_X64' '-DV8_EMBEDDER_STRING="-node.14"' '-DENABLE_DISASSEMBLER' '-DV8_PROMISE_INTERNAL_FIELD_COUNT' '-Dv8_promise_internal_field_count' '-DV8_INTL_SUPPORT' '-DV8_CONCURRENT_MARKING' '-DDISABLE_UNTRUSTED_CODE_MITIGATIONS' '-DICU_UTIL_DATA_IMPL=ICU_UTIL_DATA_STATIC' '-DUCONFIG_NO_SERVICE=1' '-DUCONFIG_NO_REGULAR_EXPRESSIONS=1' '-DU_ENABLE_DYLOAD=0' '-DU_STATIC_IMPLEMENTATION=1' '-DU_HAVE_STD_STRING=1' '-DUCONFIG_NO_BREAK_ITERATION=0' -I../deps/v8 -I../. -I/private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj/gen -I/nix/store/j54sy9zx13nmkcbic1g1dyhha4h1nnwl-zlib-1.2.11-dev/include -I/nix/store/1gadgxvbf1xdiffs9p9fz8xvb8hwx7ia-libuv-1.21.0/include -I/nix/store/b069mlwx07cm18p53y69p4asr2i81dym-openssl-1.1.1-dev/include -I../deps/v8/include -I../deps/icu-small/source/i18n -I../deps/icu-small/source/common  -O3 -gdwarf-2 -fstrict-aliasing -mmacosx-version-min=10.7 -arch x86_64 -Wall -Wendif-labels -W -Wno-unused-parameter -std=gnu++1y -stdlib=libc++ -fno-rtti -fno-exceptions -fno-threadsafe-statics -fno-strict-aliasing -MMD -MF /private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/.deps//private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj.target/v8_base/deps/v8/src/snapshot/startup-serializer.o.d.raw   -c -o /private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj.target/v8_base/deps/v8/src/snapshot/startup-serializer.o ../deps/v8/src/snapshot/startup-serializer.cc
  clang++ '-DV8_GYP_BUILD' '-DV8_TYPED_ARRAY_MAX_SIZE_IN_HEAP=0' '-D_DARWIN_USE_64_BIT_INODE=1' '-DV8_TARGET_ARCH_X64' '-DV8_EMBEDDER_STRING="-node.14"' '-DENABLE_DISASSEMBLER' '-DV8_PROMISE_INTERNAL_FIELD_COUNT' '-Dv8_promise_internal_field_count' '-DV8_INTL_SUPPORT' '-DV8_CONCURRENT_MARKING' '-DDISABLE_UNTRUSTED_CODE_MITIGATIONS' '-DICU_UTIL_DATA_IMPL=ICU_UTIL_DATA_STATIC' '-DUCONFIG_NO_SERVICE=1' '-DUCONFIG_NO_REGULAR_EXPRESSIONS=1' '-DU_ENABLE_DYLOAD=0' '-DU_STATIC_IMPLEMENTATION=1' '-DU_HAVE_STD_STRING=1' '-DUCONFIG_NO_BREAK_ITERATION=0' -I../deps/v8 -I../. -I/private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj/gen -I/nix/store/j54sy9zx13nmkcbic1g1dyhha4h1nnwl-zlib-1.2.11-dev/include -I/nix/store/1gadgxvbf1xdiffs9p9fz8xvb8hwx7ia-libuv-1.21.0/include -I/nix/store/b069mlwx07cm18p53y69p4asr2i81dym-openssl-1.1.1-dev/include -I../deps/v8/include -I../deps/icu-small/source/i18n -I../deps/icu-small/source/common  -O3 -gdwarf-2 -fstrict-aliasing -mmacosx-version-min=10.7 -arch x86_64 -Wall -Wendif-labels -W -Wno-unused-parameter -std=gnu++1y -stdlib=libc++ -fno-rtti -fno-exceptions -fno-threadsafe-statics -fno-strict-aliasing -MMD -MF /private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/.deps//private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj.target/v8_base/deps/v8/src/source-position-table.o.d.raw   -c -o /private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj.target/v8_base/deps/v8/src/source-position-table.o ../deps/v8/src/source-position-table.cc
  clang++ '-DV8_GYP_BUILD' '-DV8_TYPED_ARRAY_MAX_SIZE_IN_HEAP=0' '-D_DARWIN_USE_64_BIT_INODE=1' '-DV8_TARGET_ARCH_X64' '-DV8_EMBEDDER_STRING="-node.14"' '-DENABLE_DISASSEMBLER' '-DV8_PROMISE_INTERNAL_FIELD_COUNT' '-Dv8_promise_internal_field_count' '-DV8_INTL_SUPPORT' '-DV8_CONCURRENT_MARKING' '-DDISABLE_UNTRUSTED_CODE_MITIGATIONS' '-DICU_UTIL_DATA_IMPL=ICU_UTIL_DATA_STATIC' '-DUCONFIG_NO_SERVICE=1' '-DUCONFIG_NO_REGULAR_EXPRESSIONS=1' '-DU_ENABLE_DYLOAD=0' '-DU_STATIC_IMPLEMENTATION=1' '-DU_HAVE_STD_STRING=1' '-DUCONFIG_NO_BREAK_ITERATION=0' -I../deps/v8 -I../. -I/private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj/gen -I/nix/store/j54sy9zx13nmkcbic1g1dyhha4h1nnwl-zlib-1.2.11-dev/include -I/nix/store/1gadgxvbf1xdiffs9p9fz8xvb8hwx7ia-libuv-1.21.0/include -I/nix/store/b069mlwx07cm18p53y69p4asr2i81dym-openssl-1.1.1-dev/include -I../deps/v8/include -I../deps/icu-small/source/i18n -I../deps/icu-small/source/common  -O3 -gdwarf-2 -fstrict-aliasing -mmacosx-version-min=10.7 -arch x86_64 -Wall -Wendif-labels -W -Wno-unused-parameter -std=gnu++1y -stdlib=libc++ -fno-rtti -fno-exceptions -fno-threadsafe-statics -fno-strict-aliasing -MMD -MF /private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/.deps//private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj.target/v8_base/deps/v8/src/source-position.o.d.raw   -c -o /private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj.target/v8_base/deps/v8/src/source-position.o ../deps/v8/src/source-position.cc
building of '/nix/store/biy5xczq4bcpzyg5pxx1y09rqchm52cc-nodejs-10.9.0.drv' timed out after 1800 seconds
error: build of '/nix/store/biy5xczq4bcpzyg5pxx1y09rqchm52cc-nodejs-10.9.0.drv' failed
```
</p></details>

orivej commented on pkgs/top-level/all-packages.nix:
```diff
@@ -11454,7 +11456,7 @@ with pkgs;
     })
     openssl_1_0_2
-    openssl_1_1_0;
+    openssl_1_1_1;
```
I think we should not introduce `openssl_1_1_1` attribute, we should rename `openssl_1_1_0` to `openssl_1_1` and use that.

alyssais replied:
Same with `openssl_1_0`?

orivej replied:
No, it does not have to be changed in this PR, and it may be left as is: AFAIK `openssl_1_0_1` and `openssl_1_0_2` were incompatible.

orivej replied:
Please don't introduce `openssl_1_0` alias for `openssl_1_0_2` yet.

globin replied:
Yes, please use `openssl_1_1` and `openssl_1_0_2`.

timokau replied:
I don't feel strongly either way, but if future 1.1.x versions are expected to break backwards compatibility to 1.1.1, having that version explicitly makes sense to me.

orivej commented on pkgs/tools/misc/powerline-rs/default.nix:
```diff
@@ -1,4 +1,4 @@
-{ stdenv, lib, rustPlatform, fetchFromGitHub, pkgconfig, file, perl, curl, cmake, openssl_1_1_0, libssh2, libgit2, libzip, Security }:
+{ stdenv, lib, rustPlatform, fetchFromGitHub, pkgconfig, file, perl, curl, cmake, openssl_1_1, libssh2, libgit2, libzip, Security }:
```
Could you use `openssl` here, and call `powerline-rs` with `openssl = openssl_1_1` in `all-packages.nix`?

globin commented:
Other than the change to `openssl_1_1` I think it's good to go since `nixpkgs` does not generally use this yet and that would be a large and complex change, for which I will hopefully have time after the release.

timokau commented:
The ofBorg failure is probably unrelated. It would be nice to still fix them in a separate commit though. They look like sandbox issues (`/usr/bin/env` is impure and not available in a sandbox) that should be fixable with a simple `patchShebangs` or `substituteInPlace`.

alyssais committed:
openssl: choose names openssl_1_0_2, openssl_1_1

alyssais committed:
powerline-rs: specify openssl version in override

orivej commented on pkgs/tools/misc/powerline-rs/default.nix:
```diff
@@ -15,7 +15,7 @@ rustPlatform.buildRustPackage rec {
 
   nativeBuildInputs = [ pkgconfig file perl cmake curl ];
-  buildInputs = [ openssl_1_1_0 libssh2 libgit2 libzip ] ++ lib.optional stdenv.isDarwin Security;
+  buildInputs = [ openssl_1_1 libssh2 libgit2 libzip ] ++ lib.optional stdenv.isDarwin Security;
```
Please replace this with `openssl` too.

fpletz committed:
openssl: fix 1.1.1 build with sandboxing enabled

fpletz commented:
@GrahamcOfBorg build ldns nodejs-10_x tor powerline-rs

GrahamcOfBorg commented:
<!--REQUEST_ID=654b36aa-2d1b-4565-bec0-9549a947494e-->
Timed out, unknown build status on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.46524&attempt_id=d9a18716-5b86-4a74-9780-57ab85f9c5b4)

Attempted: nodejs-10_x, powerline-rs

The following builds were skipped because they don't evaluate on x86_64-darwin: ldns, tor

<details><summary>Partial log (click to expand)</summary><p>

```
  clang++ '-DV8_GYP_BUILD' '-DV8_TYPED_ARRAY_MAX_SIZE_IN_HEAP=0' '-D_DARWIN_USE_64_BIT_INODE=1' '-DV8_TARGET_ARCH_X64' '-DV8_EMBEDDER_STRING="-node.14"' '-DENABLE_DISASSEMBLER' '-DV8_PROMISE_INTERNAL_FIELD_COUNT' '-Dv8_promise_internal_field_count' '-DV8_INTL_SUPPORT' '-DV8_CONCURRENT_MARKING' '-DDISABLE_UNTRUSTED_CODE_MITIGATIONS' '-DICU_UTIL_DATA_IMPL=ICU_UTIL_DATA_STATIC' '-DUCONFIG_NO_SERVICE=1' '-DUCONFIG_NO_REGULAR_EXPRESSIONS=1' '-DU_ENABLE_DYLOAD=0' '-DU_STATIC_IMPLEMENTATION=1' '-DU_HAVE_STD_STRING=1' '-DUCONFIG_NO_BREAK_ITERATION=0' -I../deps/v8 -I../. -I/private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj/gen -I/nix/store/j54sy9zx13nmkcbic1g1dyhha4h1nnwl-zlib-1.2.11-dev/include -I/nix/store/1gadgxvbf1xdiffs9p9fz8xvb8hwx7ia-libuv-1.21.0/include -I/nix/store/gpw7q363dmc5nxp8asl9rbgblk4gfmyw-openssl-1.1.1-dev/include -I../deps/v8/include -I../deps/icu-small/source/i18n -I../deps/icu-small/source/common  -O3 -gdwarf-2 -fstrict-aliasing -mmacosx-version-min=10.7 -arch x86_64 -Wall -Wendif-labels -W -Wno-unused-parameter -std=gnu++1y -stdlib=libc++ -fno-rtti -fno-exceptions -fno-threadsafe-statics -fno-strict-aliasing -MMD -MF /private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/.deps//private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj.target/v8_base/deps/v8/src/snapshot/builtin-serializer-allocator.o.d.raw   -c -o /private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj.target/v8_base/deps/v8/src/snapshot/builtin-serializer-allocator.o ../deps/v8/src/snapshot/builtin-serializer-allocator.cc
  clang++ '-DV8_GYP_BUILD' '-DV8_TYPED_ARRAY_MAX_SIZE_IN_HEAP=0' '-D_DARWIN_USE_64_BIT_INODE=1' '-DV8_TARGET_ARCH_X64' '-DV8_EMBEDDER_STRING="-node.14"' '-DENABLE_DISASSEMBLER' '-DV8_PROMISE_INTERNAL_FIELD_COUNT' '-Dv8_promise_internal_field_count' '-DV8_INTL_SUPPORT' '-DV8_CONCURRENT_MARKING' '-DDISABLE_UNTRUSTED_CODE_MITIGATIONS' '-DICU_UTIL_DATA_IMPL=ICU_UTIL_DATA_STATIC' '-DUCONFIG_NO_SERVICE=1' '-DUCONFIG_NO_REGULAR_EXPRESSIONS=1' '-DU_ENABLE_DYLOAD=0' '-DU_STATIC_IMPLEMENTATION=1' '-DU_HAVE_STD_STRING=1' '-DUCONFIG_NO_BREAK_ITERATION=0' -I../deps/v8 -I../. -I/private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj/gen -I/nix/store/j54sy9zx13nmkcbic1g1dyhha4h1nnwl-zlib-1.2.11-dev/include -I/nix/store/1gadgxvbf1xdiffs9p9fz8xvb8hwx7ia-libuv-1.21.0/include -I/nix/store/gpw7q363dmc5nxp8asl9rbgblk4gfmyw-openssl-1.1.1-dev/include -I../deps/v8/include -I../deps/icu-small/source/i18n -I../deps/icu-small/source/common  -O3 -gdwarf-2 -fstrict-aliasing -mmacosx-version-min=10.7 -arch x86_64 -Wall -Wendif-labels -W -Wno-unused-parameter -std=gnu++1y -stdlib=libc++ -fno-rtti -fno-exceptions -fno-threadsafe-statics -fno-strict-aliasing -MMD -MF /private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/.deps//private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj.target/v8_base/deps/v8/src/snapshot/builtin-serializer.o.d.raw   -c -o /private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj.target/v8_base/deps/v8/src/snapshot/builtin-serializer.o ../deps/v8/src/snapshot/builtin-serializer.cc
  clang++ '-DV8_GYP_BUILD' '-DV8_TYPED_ARRAY_MAX_SIZE_IN_HEAP=0' '-D_DARWIN_USE_64_BIT_INODE=1' '-DV8_TARGET_ARCH_X64' '-DV8_EMBEDDER_STRING="-node.14"' '-DENABLE_DISASSEMBLER' '-DV8_PROMISE_INTERNAL_FIELD_COUNT' '-Dv8_promise_internal_field_count' '-DV8_INTL_SUPPORT' '-DV8_CONCURRENT_MARKING' '-DDISABLE_UNTRUSTED_CODE_MITIGATIONS' '-DICU_UTIL_DATA_IMPL=ICU_UTIL_DATA_STATIC' '-DUCONFIG_NO_SERVICE=1' '-DUCONFIG_NO_REGULAR_EXPRESSIONS=1' '-DU_ENABLE_DYLOAD=0' '-DU_STATIC_IMPLEMENTATION=1' '-DU_HAVE_STD_STRING=1' '-DUCONFIG_NO_BREAK_ITERATION=0' -I../deps/v8 -I../. -I/private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj/gen -I/nix/store/j54sy9zx13nmkcbic1g1dyhha4h1nnwl-zlib-1.2.11-dev/include -I/nix/store/1gadgxvbf1xdiffs9p9fz8xvb8hwx7ia-libuv-1.21.0/include -I/nix/store/gpw7q363dmc5nxp8asl9rbgblk4gfmyw-openssl-1.1.1-dev/include -I../deps/v8/include -I../deps/icu-small/source/i18n -I../deps/icu-small/source/common  -O3 -gdwarf-2 -fstrict-aliasing -mmacosx-version-min=10.7 -arch x86_64 -Wall -Wendif-labels -W -Wno-unused-parameter -std=gnu++1y -stdlib=libc++ -fno-rtti -fno-exceptions -fno-threadsafe-statics -fno-strict-aliasing -MMD -MF /private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/.deps//private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj.target/v8_base/deps/v8/src/snapshot/builtin-snapshot-utils.o.d.raw   -c -o /private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj.target/v8_base/deps/v8/src/snapshot/builtin-snapshot-utils.o ../deps/v8/src/snapshot/builtin-snapshot-utils.cc
  clang++ '-DV8_GYP_BUILD' '-DV8_TYPED_ARRAY_MAX_SIZE_IN_HEAP=0' '-D_DARWIN_USE_64_BIT_INODE=1' '-DV8_TARGET_ARCH_X64' '-DV8_EMBEDDER_STRING="-node.14"' '-DENABLE_DISASSEMBLER' '-DV8_PROMISE_INTERNAL_FIELD_COUNT' '-Dv8_promise_internal_field_count' '-DV8_INTL_SUPPORT' '-DV8_CONCURRENT_MARKING' '-DDISABLE_UNTRUSTED_CODE_MITIGATIONS' '-DICU_UTIL_DATA_IMPL=ICU_UTIL_DATA_STATIC' '-DUCONFIG_NO_SERVICE=1' '-DUCONFIG_NO_REGULAR_EXPRESSIONS=1' '-DU_ENABLE_DYLOAD=0' '-DU_STATIC_IMPLEMENTATION=1' '-DU_HAVE_STD_STRING=1' '-DUCONFIG_NO_BREAK_ITERATION=0' -I../deps/v8 -I../. -I/private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj/gen -I/nix/store/j54sy9zx13nmkcbic1g1dyhha4h1nnwl-zlib-1.2.11-dev/include -I/nix/store/1gadgxvbf1xdiffs9p9fz8xvb8hwx7ia-libuv-1.21.0/include -I/nix/store/gpw7q363dmc5nxp8asl9rbgblk4gfmyw-openssl-1.1.1-dev/include -I../deps/v8/include -I../deps/icu-small/source/i18n -I../deps/icu-small/source/common  -O3 -gdwarf-2 -fstrict-aliasing -mmacosx-version-min=10.7 -arch x86_64 -Wall -Wendif-labels -W -Wno-unused-parameter -std=gnu++1y -stdlib=libc++ -fno-rtti -fno-exceptions -fno-threadsafe-statics -fno-strict-aliasing -MMD -MF /private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/.deps//private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj.target/v8_base/deps/v8/src/snapshot/code-serializer.o.d.raw   -c -o /private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj.target/v8_base/deps/v8/src/snapshot/code-serializer.o ../deps/v8/src/snapshot/code-serializer.cc
  clang++ '-DV8_GYP_BUILD' '-DV8_TYPED_ARRAY_MAX_SIZE_IN_HEAP=0' '-D_DARWIN_USE_64_BIT_INODE=1' '-DV8_TARGET_ARCH_X64' '-DV8_EMBEDDER_STRING="-node.14"' '-DENABLE_DISASSEMBLER' '-DV8_PROMISE_INTERNAL_FIELD_COUNT' '-Dv8_promise_internal_field_count' '-DV8_INTL_SUPPORT' '-DV8_CONCURRENT_MARKING' '-DDISABLE_UNTRUSTED_CODE_MITIGATIONS' '-DICU_UTIL_DATA_IMPL=ICU_UTIL_DATA_STATIC' '-DUCONFIG_NO_SERVICE=1' '-DUCONFIG_NO_REGULAR_EXPRESSIONS=1' '-DU_ENABLE_DYLOAD=0' '-DU_STATIC_IMPLEMENTATION=1' '-DU_HAVE_STD_STRING=1' '-DUCONFIG_NO_BREAK_ITERATION=0' -I../deps/v8 -I../. -I/private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj/gen -I/nix/store/j54sy9zx13nmkcbic1g1dyhha4h1nnwl-zlib-1.2.11-dev/include -I/nix/store/1gadgxvbf1xdiffs9p9fz8xvb8hwx7ia-libuv-1.21.0/include -I/nix/store/gpw7q363dmc5nxp8asl9rbgblk4gfmyw-openssl-1.1.1-dev/include -I../deps/v8/include -I../deps/icu-small/source/i18n -I../deps/icu-small/source/common  -O3 -gdwarf-2 -fstrict-aliasing -mmacosx-version-min=10.7 -arch x86_64 -Wall -Wendif-labels -W -Wno-unused-parameter -std=gnu++1y -stdlib=libc++ -fno-rtti -fno-exceptions -fno-threadsafe-statics -fno-strict-aliasing -MMD -MF /private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/.deps//private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj.target/v8_base/deps/v8/src/snapshot/default-deserializer-allocator.o.d.raw   -c -o /private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj.target/v8_base/deps/v8/src/snapshot/default-deserializer-allocator.o ../deps/v8/src/snapshot/default-deserializer-allocator.cc
  clang++ '-DV8_GYP_BUILD' '-DV8_TYPED_ARRAY_MAX_SIZE_IN_HEAP=0' '-D_DARWIN_USE_64_BIT_INODE=1' '-DV8_TARGET_ARCH_X64' '-DV8_EMBEDDER_STRING="-node.14"' '-DENABLE_DISASSEMBLER' '-DV8_PROMISE_INTERNAL_FIELD_COUNT' '-Dv8_promise_internal_field_count' '-DV8_INTL_SUPPORT' '-DV8_CONCURRENT_MARKING' '-DDISABLE_UNTRUSTED_CODE_MITIGATIONS' '-DICU_UTIL_DATA_IMPL=ICU_UTIL_DATA_STATIC' '-DUCONFIG_NO_SERVICE=1' '-DUCONFIG_NO_REGULAR_EXPRESSIONS=1' '-DU_ENABLE_DYLOAD=0' '-DU_STATIC_IMPLEMENTATION=1' '-DU_HAVE_STD_STRING=1' '-DUCONFIG_NO_BREAK_ITERATION=0' -I../deps/v8 -I../. -I/private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj/gen -I/nix/store/j54sy9zx13nmkcbic1g1dyhha4h1nnwl-zlib-1.2.11-dev/include -I/nix/store/1gadgxvbf1xdiffs9p9fz8xvb8hwx7ia-libuv-1.21.0/include -I/nix/store/gpw7q363dmc5nxp8asl9rbgblk4gfmyw-openssl-1.1.1-dev/include -I../deps/v8/include -I../deps/icu-small/source/i18n -I../deps/icu-small/source/common  -O3 -gdwarf-2 -fstrict-aliasing -mmacosx-version-min=10.7 -arch x86_64 -Wall -Wendif-labels -W -Wno-unused-parameter -std=gnu++1y -stdlib=libc++ -fno-rtti -fno-exceptions -fno-threadsafe-statics -fno-strict-aliasing -MMD -MF /private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/.deps//private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj.target/v8_base/deps/v8/src/snapshot/default-serializer-allocator.o.d.raw   -c -o /private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj.target/v8_base/deps/v8/src/snapshot/default-serializer-allocator.o ../deps/v8/src/snapshot/default-serializer-allocator.cc
  clang++ '-DV8_GYP_BUILD' '-DV8_TYPED_ARRAY_MAX_SIZE_IN_HEAP=0' '-D_DARWIN_USE_64_BIT_INODE=1' '-DV8_TARGET_ARCH_X64' '-DV8_EMBEDDER_STRING="-node.14"' '-DENABLE_DISASSEMBLER' '-DV8_PROMISE_INTERNAL_FIELD_COUNT' '-Dv8_promise_internal_field_count' '-DV8_INTL_SUPPORT' '-DV8_CONCURRENT_MARKING' '-DDISABLE_UNTRUSTED_CODE_MITIGATIONS' '-DICU_UTIL_DATA_IMPL=ICU_UTIL_DATA_STATIC' '-DUCONFIG_NO_SERVICE=1' '-DUCONFIG_NO_REGULAR_EXPRESSIONS=1' '-DU_ENABLE_DYLOAD=0' '-DU_STATIC_IMPLEMENTATION=1' '-DU_HAVE_STD_STRING=1' '-DUCONFIG_NO_BREAK_ITERATION=0' -I../deps/v8 -I../. -I/private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj/gen -I/nix/store/j54sy9zx13nmkcbic1g1dyhha4h1nnwl-zlib-1.2.11-dev/include -I/nix/store/1gadgxvbf1xdiffs9p9fz8xvb8hwx7ia-libuv-1.21.0/include -I/nix/store/gpw7q363dmc5nxp8asl9rbgblk4gfmyw-openssl-1.1.1-dev/include -I../deps/v8/include -I../deps/icu-small/source/i18n -I../deps/icu-small/source/common  -O3 -gdwarf-2 -fstrict-aliasing -mmacosx-version-min=10.7 -arch x86_64 -Wall -Wendif-labels -W -Wno-unused-parameter -std=gnu++1y -stdlib=libc++ -fno-rtti -fno-exceptions -fno-threadsafe-statics -fno-strict-aliasing -MMD -MF /private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/.deps//private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj.target/v8_base/deps/v8/src/snapshot/deserializer.o.d.raw   -c -o /private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj.target/v8_base/deps/v8/src/snapshot/deserializer.o ../deps/v8/src/snapshot/deserializer.cc
  clang++ '-DV8_GYP_BUILD' '-DV8_TYPED_ARRAY_MAX_SIZE_IN_HEAP=0' '-D_DARWIN_USE_64_BIT_INODE=1' '-DV8_TARGET_ARCH_X64' '-DV8_EMBEDDER_STRING="-node.14"' '-DENABLE_DISASSEMBLER' '-DV8_PROMISE_INTERNAL_FIELD_COUNT' '-Dv8_promise_internal_field_count' '-DV8_INTL_SUPPORT' '-DV8_CONCURRENT_MARKING' '-DDISABLE_UNTRUSTED_CODE_MITIGATIONS' '-DICU_UTIL_DATA_IMPL=ICU_UTIL_DATA_STATIC' '-DUCONFIG_NO_SERVICE=1' '-DUCONFIG_NO_REGULAR_EXPRESSIONS=1' '-DU_ENABLE_DYLOAD=0' '-DU_STATIC_IMPLEMENTATION=1' '-DU_HAVE_STD_STRING=1' '-DUCONFIG_NO_BREAK_ITERATION=0' -I../deps/v8 -I../. -I/private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj/gen -I/nix/store/j54sy9zx13nmkcbic1g1dyhha4h1nnwl-zlib-1.2.11-dev/include -I/nix/store/1gadgxvbf1xdiffs9p9fz8xvb8hwx7ia-libuv-1.21.0/include -I/nix/store/gpw7q363dmc5nxp8asl9rbgblk4gfmyw-openssl-1.1.1-dev/include -I../deps/v8/include -I../deps/icu-small/source/i18n -I../deps/icu-small/source/common  -O3 -gdwarf-2 -fstrict-aliasing -mmacosx-version-min=10.7 -arch x86_64 -Wall -Wendif-labels -W -Wno-unused-parameter -std=gnu++1y -stdlib=libc++ -fno-rtti -fno-exceptions -fno-threadsafe-statics -fno-strict-aliasing -MMD -MF /private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/.deps//private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj.target/v8_base/deps/v8/src/snapshot/natives-common.o.d.raw   -c -o /private/tmp/nix-build-nodejs-10.9.0.drv-0/node-v10.9.0/out/Release/obj.target/v8_base/deps/v8/src/snapshot/natives-common.o ../deps/v8/src/snapshot/natives-common.cc
building of '/nix/store/5xca4i0p623s0nhvsqnc8g0d6m0m2v7q-nodejs-10.9.0.drv' timed out after 1800 seconds
error: build of '/nix/store/5xca4i0p623s0nhvsqnc8g0d6m0m2v7q-nodejs-10.9.0.drv' failed
```
</p></details>

fpletz commented:
nodejs-10_x builds locally for me. Seems to be a timeout issue but doesn't make much sense according to the log.

@grahamc Is there a timeout per package?

timokau commented:
No, the timeout is per build including all dependencies.

fpletz commented:
Then the question is how following can be possible ([from the x86_64-linux log](https://logs.nix.ci/?key=nixos/nixpkgs.46524&attempt_id=14b20dd2-f04d-4264-a071-759883c06ecb#16503)):

```
installing /nix/store/dvs7h6kgb3qrg9dxjvq93fzqbng5sjbd-nodejs-10.9.0/lib/node_modules/npm/node_modules/tweetnacl/CHANGELOG.md
installing /nix/store/dvs7h6kgb3qrg9dxjvq93fzqbng5sjbd-nodejs-10.9.0/lib/node_modules/npm/node_modules/tweetnacl/PULL_REQUEST_TEMPLATE.md
installing /nix/store/dvs7h6kgb3qrg9dxjvq93fzqbng5sjbd-nodejs-10.9.0/lib/node_modules/npm/node_modules/tweetnacl/AUTHORS.md
installing /nix/store/dvs7h6kgb3qrg9dxjvq93fzqbng5sjbd-nodejs-10.9.0/lib/node_modules/npm/node_modules/tweetnacl/README.md
installing /nix/store/dvs7h6kgb3qrg9dxjvq93fzqbng5s
builder for '/nix/store/dgilcdiprs124kwwm6vgxkb1ylwg70vm-nodejs-10.9.0.drv' failed with exit code 2
building '/nix/store/3z707vqg4k2biwlljdy070xpbdf21si0-powerline-rs-0.1.8.drv'...
unpacking sources
unpacking source archive /nix/store/bvklx0kpx3giaqcvbrb2r7i9mk6h4z6k-source
source root is source
```

GrahamcOfBorg commented:
<!--REQUEST_ID=15d31592-3ef1-42c4-b319-405fad367a97-->
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.46524&attempt_id=7bc22e15-8455-452b-a472-18d3426e9dba)

Attempted: tor

<details><summary>Partial log (click to expand)</summary><p>

```
strip is /nix/store/h0lbngpv6ln56hjj59i6l77vxq25flbz-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/8i1893wiw5gp778qgg03d3mqy3zah15h-tor-0.3.4.8/bin
patching script interpreter paths in /nix/store/8i1893wiw5gp778qgg03d3mqy3zah15h-tor-0.3.4.8
/nix/store/8i1893wiw5gp778qgg03d3mqy3zah15h-tor-0.3.4.8/bin/torify: interpreter directive changed from " /bin/sh" to "/nix/store/czx8vkrb9jdgjyz8qfksh10vrnqa723l-bash-4.4-p23/bin/sh"
checking for references to /build in /nix/store/8i1893wiw5gp778qgg03d3mqy3zah15h-tor-0.3.4.8...
shrinking RPATHs of ELF executables and libraries in /nix/store/w0csdvqxswvbwn8mp6ajs0h3r3xk9xz9-tor-0.3.4.8-geoip
strip is /nix/store/h0lbngpv6ln56hjj59i6l77vxq25flbz-binutils-2.30/bin/strip
patching script interpreter paths in /nix/store/w0csdvqxswvbwn8mp6ajs0h3r3xk9xz9-tor-0.3.4.8-geoip
checking for references to /build in /nix/store/w0csdvqxswvbwn8mp6ajs0h3r3xk9xz9-tor-0.3.4.8-geoip...
/nix/store/8i1893wiw5gp778qgg03d3mqy3zah15h-tor-0.3.4.8
```
</p></details>

orivej commented:
@GrahamcOfBorg build openssl_1_1

GrahamcOfBorg commented:
<!--REQUEST_ID=3ea0b411-6095-42ff-9f3c-c81414093684-->
Success on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.46524&attempt_id=077196f6-0d5b-459b-ba8e-cb8a81ecea64)

Attempted: openssl_1_1

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/al9bhp78cmwx3nyfym30k5a8w5fx34mp-openssl-1.1.1-bin
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=3ea0b411-6095-42ff-9f3c-c81414093684-->
Success on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.46524&attempt_id=eed86d86-6406-420c-bad6-07edb0b74a02)

Attempted: openssl_1_1

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/7wq9rshkg2j3mc9fy1pn7jcg7q19kydn-openssl-1.1.1-bin
```
</p></details>

GrahamcOfBorg commented:
<!--REQUEST_ID=3ea0b411-6095-42ff-9f3c-c81414093684-->
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.46524&attempt_id=d2552858-3c8a-413e-8614-e290b476f6f7)

Attempted: openssl_1_1

<details><summary>Partial log (click to expand)</summary><p>

```
/nix/store/ynvzqq6w622jlj2mdgh7r2qrq0315s4s-openssl-1.1.1-bin
```
</p></details>
